src/openrct2/rct2/S6Importer.cpp

```diff
@@ -800,7 +800,12 @@ class S6Importer final : public IParkImporter
             }
         }
 
-        dst->music = src->music;
+        auto musicStyle = OBJECT_ENTRY_INDEX_NULL;
+        if (GetRideTypeDescriptor(dst->type).HasFlag(RIDE_TYPE_FLAG_ALLOW_MUSIC))
+        {
+            musicStyle = src->music;
+        }
+        dst->music = musicStyle;
```

Comment:
I dont see the reverse in S6Exporter. I assume its not required as it truncates fine?